### PR TITLE
Add a separate polygon agent

### PR DESCRIPTION
**Pull request by dgusakov**

### CODE REVIEW

File: polygon/src/agent-dao-ops.ts [hunk 1]
```diff
@@ -0,0 +1,136 @@
...
+    }
+
+    if (
+      bufferedMatic.isGreaterThanOrEqualTo(
+        totalPooledMatic.div(100).times(MAX_BUFFERED_MATIC_IMMEDIATE_PERCENT)
+      )
+    ) {
+      findings.push(
+        Finding.fromObject({
+          name: "Huge buffered ETH amount",
```
Comment (madlabman) [inline polygon/src/agent-dao-ops.ts:106]:
```suggestion
          name: "Huge buffered MATIC amount",
```

File: polygon/src/agent-dao-ops.ts [hunk 2]
```diff
@@ -0,0 +1,136 @@
...
+          description: `There are ${bufferedMatic.toFixed(
+            4
+          )} (more that ${MAX_BUFFERED_MATIC_IMMEDIATE_PERCENT}% of total pooled MATIC) buffered MATIC in stMATIC contract`,
```
Comment (madlabman) [inline polygon/src/agent-dao-ops.ts:109]:
```suggestion
          )} (more than ${MAX_BUFFERED_MATIC_IMMEDIATE_PERCENT}% of total pooled MATIC) buffered MATIC in stMATIC contract`,
```

File: polygon/src/agent-dao-ops.ts [hunk 3]
```diff
@@ -0,0 +1,136 @@
...
+        })
+      );
+      lastReportedBufferedMatic = now;
+    } else if (
+      timeHighPooledMaticStart != 0 &&
+      now - timeHighPooledMaticStart > FULL_24_HOURS
+    ) {
+      findings.push(
+        Finding.fromObject({
+          name: "High buffered ETH amount",
```
Comment (madlabman) [inline polygon/src/agent-dao-ops.ts:122]:
```suggestion
          name: "High buffered MATIC amount",
```

File: polygon/src/agent-dao-ops.ts [hunk 4]
```diff
@@ -0,0 +1,136 @@
...
+          description: `There are ${bufferedMatic.toFixed(
+            4
+          )} (more that ${MAX_BUFFERED_MATIC_DAILY_PERCENT}% of total pooled MATIC) buffered MATIC in stMATIC contract for more than ${Math.floor(
```
Comment (madlabman) [inline polygon/src/agent-dao-ops.ts:125]:
```suggestion
          )} (more than ${MAX_BUFFERED_MATIC_DAILY_PERCENT}% of total pooled MATIC) buffered MATIC in stMATIC contract for more than ${Math.floor(
```

**Review by madlabman — CHANGES_REQUESTED** [review]

File: polygon/src/agent-dao-ops.ts (hunk 1 above)
Comment (dgusakov) [reply polygon/src/agent-dao-ops.ts]: Fixed

File: polygon/src/agent-dao-ops.ts (hunk 2 above)
Comment (dgusakov) [reply polygon/src/agent-dao-ops.ts]: Fixed

File: polygon/src/agent-dao-ops.ts (hunk 3 above)
Comment (dgusakov) [reply polygon/src/agent-dao-ops.ts]: Fixed

File: polygon/src/agent-dao-ops.ts (hunk 4 above)
Comment (dgusakov) [reply polygon/src/agent-dao-ops.ts]: Fixed

File: polygon/src/agent-dao-ops.ts [hunk 5]
```diff
@@ -0,0 +1,136 @@
...
+
+let lastReportedBufferedMatic = 0;
+let timeHighPooledMaticStart = 0;
+
+export async function initialize(
+  currentBlock: number
+): Promise<{ [key: string]: string }> {
+  console.log(`[${name}]`);
+  return {};
+}
```
Comment (vgorkavenko) [inline polygon/src/agent-dao-ops.ts:38]: Why is `currentBlock` here? Maybe it should also be output?

**Review by vgorkavenko — APPROVED** [review]

File: polygon/src/agent-dao-ops.ts (hunk 5 above)
Comment (dgusakov) [reply polygon/src/agent-dao-ops.ts:38]: Nope. initialize can be more complicated and include multiple rpc calls

**Pull request merged**